library/core/tests/mem.rs

```diff
@@ -161,22 +161,6 @@ fn assume_init_good() {
     assert!(TRUE);
 }
 
-#[test]
-fn uninit_array_assume_init() {
```

Comment:
Is there an equivalent test for `transpose` already?

Reply:
Maybe? There are doc tests, but again transpose is NOT assume_init so I'm not sure what we'd be testing (that transmute works?).

Reply:
We'd be testing that this same behavior still works with the new methods, which sounds useful to me.